contracts/PetPark.sol

```diff
@@ -1,7 +1,92 @@
 //SPDX-License-Identifier: Unlicense
 pragma solidity ^0.8.0;
+/// @title Assignment-1 Smart Contract for Pet Park
+/// @author Srigowri M V
+/// @dev The functions implemented pass the tests provided in /test/PetPark.spec.js
+contract PetPark {
+    //Constants
+    enum AnimalType{ None, Fish, Cat, Dog, Rabbit,Parrot}
+    enum GenderType {Male, Female}
 
+    //Events
+    event Added(AnimalType _animalType, uint count);
+    event Borrowed(AnimalType _animalType);
+    event Returned(AnimalType _animalType);
 
-contract PetPark {
+    //State Variable Declaration
+    address public owner;
+    mapping(AnimalType => uint) public animalCounts;  // Mapping for Animal Type: Count
+    mapping(address => AnimalType) public borrowerAnimal; // Mapping for Borrower Address : Animal Type
+    mapping(address => uint) public borrowerAge;      // Mapping for BorrowerAddress : Age    
+    mapping(address => GenderType) public borrowerGender; // Mapping for BorrowerAddress : Gender
+
+    //Constructor
+    constructor(){
+        owner = msg.sender;
+    }
+    //Modifiers
+    modifier ownerOnly(){
+        //Used while adding new pets to the park
+        require(msg.sender == owner, "Not an owner");
+        _;
+    }
+    
+    modifier validPet(AnimalType _animalType){
+            //Used to ensure only valid animal types are being added or borrowed
+            bool valid = (_animalType == AnimalType.Fish || 
+                              _animalType == AnimalType.Cat ||
+                              _animalType == AnimalType.Dog || 
+                              _animalType == AnimalType.Rabbit ||
+                              _animalType == AnimalType.Parrot);
+
+            require(valid, "Invalid animal type");
+            _;
+    }
+
+    //Functions
+    function add(AnimalType _animalType, uint count) external ownerOnly() validPet(_animalType){        
+        //Increment count of the corresponding pet in the park
+        animalCounts[_animalType] += count;
+        emit Added(_animalType, count);
+    }
+
+    function giveBackAnimal() external {
+        //Can give back pets only if borrowed
+        require(borrowerAnimal[msg.sender] != AnimalType.None,  "No borrowed pets");        
```

Comment:
Since `borrowerAnimal[msg.sender]` is being used multiple times it can be stored in a single storage variable